service/share/full_availability.go

```diff
@@ -26,6 +27,17 @@ func NewFullAvailability(service format.DAGService) Availability {
 // SharesAvailable reconstructs the data committed to the given Root by requesting
 // enough Shares from the network.
 func (fa *fullAvailability) SharesAvailable(ctx context.Context, root *Root) error {
+	ctx, cancel := context.WithTimeout(ctx, AvailabilityTimeout)
+	defer cancel()
+
 	_, err := ipld.RetrieveData(ctx, root, fa.service, rsmt2d.NewRSGF8Codec())
```

Comment:
If we switch to leopard there will be a lot of places like this that will need changing. I think we should simply use the constructor in the consts package in core everywhere instead. Orthogonal to this PR though.

Reply:
Can someone file an issue to track this refactoring?

Reply:
https://github.com/celestiaorg/celestia-node/issues/604